ckanext/canada/helpers.py

```diff
@@ -949,3 +952,42 @@ def max_resources_per_dataset():
     max_resource_count = config.get('ckanext.canada.max_resources_per_dataset', None)
     if max_resource_count:
         return int(max_resource_count)
+
+
+def canada_api_tracking_info(id: str):
+    fernet_key = os.environ.get('CKAN_API_TRACKING_SECRET')
+    if not fernet_key:
+        return
+
+    tracking_obj = model.Session.query(TrackingUsage).filter(TrackingUsage.id == id).first()
+    if not tracking_obj:
+        return
+
+    return_dict = {'method': tracking_obj.extras.get('method') if tracking_obj.extras else None,
+                   'timestamp': tracking_obj.timestamp,
+                   'object_type': tracking_obj.object_type,
+                   'token_name': tracking_obj.token_name,
+                   'user_id': tracking_obj.user_id}
+
+    if tracking_obj.user_id:
+        user_obj = model.User.get(tracking_obj.user_id)
+        if user_obj:
+            return_dict['user_name'] = user_obj.name
+            return_dict['user_fullname'] = user_obj.fullname
+
+    if tracking_obj.extras and (tracking_obj.extras.get('REQUEST_ARGS') or tracking_obj.extras.get('REQUEST_BODY')):
+        fernet_key = fernet_key if isinstance(fernet_key, bytes) else fernet_key.encode()
```

Comment:
maybe `encode('ascii')` otherwise it's going to default to the system encoding, but I see that Fernet() accepts either bytes or str, so why the conversion here?

Reply:
it screamed at me that it needed bytes

Reply:
works for me?
```python
Fernet('YWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWE=')
<cryptography.fernet.Fernet object at 0x7f5d1a9fa530>
```

Reply:
ah, it is the encrypt/decrypt methods that scream for the bytes. Will remove the encode on the key